lib/bitcoin/lock/LockResolver.ts

```diff
@@ -87,15 +85,17 @@ export default class LockResolver {
     // (C). verify that the lock duration is valid
     const unlockAtBlock = lockStartBlock + scriptVerifyResult.lockDurationInBlocks!;
 
+    const lockDurationInBlocks = this.versionManager.getLockDurationInBlocks(lockStartBlock);
+
     if (!this.isLockDurationValid(lockStartBlock, unlockAtBlock)) {

```

Comment:
The logic technically hasn't changed. Before, we were still doing the same thing where lock monitor uses the MAX value in protocol parameter to lock. The only difference it that now we can get a different number depending on version.

We still need this because
1. This allows you to pass in any lock and resolve it, even if a non sidetree node wrote it. It's more general purposed. (MY personal opinion)
2. In LockMonitor, we catch this specific error to set the lock state. It is set to pending status and no active lock.